### DISCUSSION TITLE
Microsoft.NET.ILLink and Microsoft.NET.Sdk.WebAssembly.Pack NuGet Packages

### DISCUSSION THREAD

**Discussion by jgw-vsolvit** [dotnet/runtime#96427] [General]
Hi all,

Although I have not explicitly added these three NuGet packages to my project, they have appeared in my NuGet Package Manager and cannot be uninstalled...

- Microsoft.NET.ILLink.Analyzers
- Microsoft.NET.ILLink.Tasks
- Microsoft.NET.Sdk.WebAssembly.Pack

I run Visual Studio 2022 (17.8.3) and .NET 8 (8.0.100). My other teammates pulling from the same repo but running older versions of Visual Studio 2022 (17.6.5) and .NET 7 (7.0.306) do not see these NuGet packages.

Any insight into what introduced these would be appreciated. Thanks!

V/r,
--John

**Comment by maraf** [comment]
These represent parts of the build time experience delivered as nuget packages instead of directly placed in the SDK installation. Based on application type and target framework version these may or may not be implicitly referenced

**Reply by jgw-vsolvit** [reply]
Thanks! Do you know if it was .NET 8 that introduced them? The concern is that they're listed as prerelease versions in my production app...

* https://www.nuget.org/packages/Microsoft.NET.ILLink/

**Reply by maraf** [reply]
The `Microsoft.NET.Sdk.WebAssembly.Pack` was introduces in .NET 8, the other two were already in .NET 7 IMO. They all should point to version `8.0.0`. Can you try to build a sample from template and share a binlog? I assume you are building blazor app if you see `Microsoft.NET.Sdk.WebAssembly.Pack`.
